Sources/PackageLoading/PackageDescription4Loader.swift

```diff
@@ -346,8 +346,16 @@ extension PackageDependencyDescription {
                 // FIXME: SwiftPM can't handle file locations with file:// scheme so we need to
                 // strip that. We need to design a Location data structure for SwiftPM.
                 let location = String(dependencyLocation.dropFirst(filePrefix.count))
-                if location.first != "/" {
-                    throw ManifestParseError.invalidManifestFormat("file:// URLs cannot be relative, did you mean to use `.package(path:)`?", diagnosticFile: nil)
+                let hostnameComponent = location.prefix(while: { $0 != "/" })
+                guard hostnameComponent.isEmpty else {
+                  if hostnameComponent == ".." {
+                    throw ManifestParseError.invalidManifestFormat(
+                      "file:// URLs cannot be relative, did you mean to use `.package(path:)`?", diagnosticFile: nil
```

Comment:
nitpick:

```suggestion
                      "file:// URLs cannot be relative, did you mean to use '.package(path:)'?", diagnosticFile: nil
```

I think it should be single quotes instead of back ticks for consistency.

Reply:
#3625 uses backticks, so I kept them as they were. I don't mind changing it if the maintainers agree.

Reply:
I think backpacks are probably not correct and we should update everywhere